### ISSUE TITLE
trouble resolving file if resolved file is in folder with '.js' in name.

### ISSUE THREAD

**Issue by pjordaan** [jonaskello/tsconfig-paths#4]
Hi, when I enable tsconfig-paths, tsconfig-paths works very well, except when I try to resolve bignumber.js

In my tsconfig.json I have
```"*": ["*", "src/*", "test/*", "vendor/node_modules/@types/*", "vendor/node_modules/*"],```
Somewhere in the code it does ```require('bignumber.js');```

If I trace the code then in match-path.js in function tryResolve() the string replacement is done incorrectly.
```
if (packageJson && packageJson.main && fileExists(path.join(physicalPath, packageJson.main))) {
        var file = path.join(physicalPath, packageJson.main);
        return file.replace(path.extname(file), "");
}
```
variable file is resolved to vendor/node_modules/bignumber.js/bignumber.js.

the file.replace removes the first '.js', so it incorrectly resolves into "vendor/node_modules/bignumber/bignumber.js", not "vendor/node_modules/bignumber.js/bignumber"

**Comment by pjordaan** [comment]
possible solutions
```
return file.replace(new RegExp(path.extname(file).replace('.', '\\.') + '$'), "");
```

```
return path.dirname(file) + path.sep + path.basename(file, path.extname(file));
```

**Comment by jonaskello** [comment]
I'll have a look when I get some time, or maybe @Jontem can look at it. However, if you have the time a PR with a failing test would be helpful :-).

**Comment by Jontem** [comment]
I've fixed this in d6b7c538324780ccfc2608982e932c40e326d429 and published a new version(2.1.1)

**Comment by pjordaan** [comment]
thx for the quick fix!
